### ISSUE TITLE
[Enhancement] Add notification in device-SDK for changes in Device Service configurations

### ISSUE THREAD

**Issue by salmanazmat666** [edgexfoundry/device-sdk-go#589]
Whenever a device is updated, a function is called `UpdateDevice()` in the GO SDK so that the service may take appropriate actions. Similarly, when the device Service is itself updated, there should be a similar function call to notify that some changes have been made in the configuration.

This would be helpful if there some settings that are common to every device, for example, baud-rate of a serial port. In almost every setup, all of the devices connected to a serial port have same baud-rates, so it doesn't make sense to have baud-rate property/attribute for each device. We can move such settings under the `[Writable]` area of service configuration and whenever there is some change, we get a notification in the SDK and be able to take appropriate actions.

Basically we should be able to get notifications for service-level changes as well just like we get for device-level changes.

**Comment by jpwhitemn** [comment]
reviewed 10/12/20 in DS WG - agree the enhancement is a reasonable one, but for Ireland (not Hanoi).
Service will still need to do something with the notification of change (from bootstrap).  The notification will need to be propagated up to the SDK.  This is not related to notification service or device service callback - strictly about [Driver] section of configuration in configuration.toml needs to be able to change and SDK is made aware.

**Comment by hahattan** [comment]
@lenny-intel do you think #833 resolve this,
If we use [CustomConfig] section instead of strictly under [Driver] section ?

**Comment by cloudxxx8** [comment]
@iain-anderson since this is in the release backlog, may we discuss it in the next Device WG meeting?

**Comment by iain-anderson** [comment]
closed #833 with custom config
